src/lib/pgen.rs

```diff
@@ -344,3 +360,127 @@ pub fn goto1(grm: &Grammar, firsts: &HashMap<String, HashSet<String>>,
     closure1(grm, firsts, &mut goto);
     goto
 }
+
+// Haven't decided yet if I want to use a struct or not
+// struct State {
+//     id: usize,
```

Comment:
Yes, we need a way to get the state from its state number anyway, so storing the number on the state itself is not really necessary.